### PR TITLE
net: propagate the user agent from node to pool

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes https://github.com/handshake-org/hsd/pull/664

TODO:
- [x] Add comment to CHANGELOG ?

### CODE REVIEW

**Review by Falci — APPROVED** [review]

**Review by rithvikvibhu — APPROVED** [review]
LGTM, agent is set correctly (and is simpler than agent, agentComment)

File: lib/node/spvnode.js [hunk 1]
```diff
@@ -73,7 +73,8 @@ class SPVNode extends Node {
       createSocket: this.config.func('create-socket'),
       memory: this.config.bool('memory'),
       selfish: true,
-      listen: false
+      listen: false,
+      agent: this.config.str('agent')
```
Comment (nodech) [inline lib/node/spvnode.js:77]: Minor nit, maybe have this grouped with other configs ? after `memory`.

**Review by nodech — APPROVED** [review]
LGTM

File: lib/node/spvnode.js (hunk 1 above)
Comment (pinheadmz) [reply lib/node/spvnode.js]: fixed.

**Review by nodech — APPROVED** [review]

File: test/net-test.js [hunk 2]
```diff
@@ -1114,4 +1117,92 @@ describe('Net', function() {
...
+      assert.strictEqual(err, 'User agent exceeds maximum length');
+    });
+
+    it('should extend user agent with comment', async () => {
+      const pool = new Pool({
+        chain,
+        agent: 'bob-wallet:v0.9.0'
+      });
+
+      const peer = await pool.createOutbound(
```
Comment (nodech) [inline test/net-test.js:1157]: no need for `await` here. It returns {Peer} directly.

File: test/net-test.js [hunk 3]
```diff
@@ -1114,4 +1117,92 @@ describe('Net', function() {
...
+        `/${pkg.name}:${pkg.version}/bob-wallet:v0.9.0/`
+      );
+    });
+
+    it('should send default user agent', async () => {
+      const pool = new Pool({
+        chain
+      });
+
+      const peer = await pool.createOutbound(
```
Comment (nodech) [inline test/net-test.js:1185]: Same here.

**Review by nodech — APPROVED** [review]

**Pull request merged**